### PR TITLE
Prepare: openPMD-api 0.15.0-dev

### PR DESCRIPTION
**Pull request by ax3l**
Anticipate openPMD-api changes in the next major release that are already in `dev` (aka Spack `develop`):
- C++17 requirement
- drop: `mpark-variant` public dependency
- add: `toml11` private dependency

Also adds @franzpoeschel as co-maintainer for the Spack package.

- [x] depends on #29483

### CODE REVIEW

File: var/spack/repos/builtin/packages/openpmd-api/package.py
Comment (ax3l) [inline var/spack/repos/builtin/packages/openpmd-api/package.py:56]: This PR needs a rebase once #29483 was merged.

**Comment by franzpoeschel** [comment]
Looks good to me
Will approve after the rebase

**Review by adamjstewart — DISMISSED** [review]

**Comment by ax3l** [comment]
Rebased :)

**Review by adamjstewart — DISMISSED** [review]

**Comment by ax3l** [comment]
@franzpoeschel @adamjstewart ready for approval & merge :)

**Review by franzpoeschel — DISMISSED** [review]

File: var/spack/repos/builtin/packages/openpmd-api/package.py
Comment (sethrj) [inline var/spack/repos/builtin/packages/openpmd-api/package.py]:
```suggestion
        if spec.satisfies('@:0.14'):  # pre C++17 releases
```

File: var/spack/repos/builtin/packages/openpmd-api/package.py
Comment (sethrj) [inline var/spack/repos/builtin/packages/openpmd-api/package.py]: In case they add a beta version ;)
```suggestion
        if spec.satisfies('@0.15:'):
```

File: var/spack/repos/builtin/packages/openpmd-api/package.py
Comment (sethrj) [inline var/spack/repos/builtin/packages/openpmd-api/package.py]:
```suggestion
        if spec.satisfies('@:0.14'):  # pre C++17 releases
```

File: var/spack/repos/builtin/packages/openpmd-api/package.py
Comment (sethrj) [inline var/spack/repos/builtin/packages/openpmd-api/package.py]:
```suggestion
        if spec.satisfies('@:0.14'):  # pre C++17 releases
```

File: var/spack/repos/builtin/packages/openpmd-api/package.py
Comment (ax3l) [reply var/spack/repos/builtin/packages/openpmd-api/package.py]: Haha, I know me and I go straight to releases :D

**Review by sethrj — APPROVED** [review]

**Pull request merged**
